sftp/sftp.go

```diff
@@ -455,9 +498,37 @@ func (o *Object) Remote() string {
 	return o.remote
 }
 
-// Hash returns "" since SFTP (in Go or OpenSSH) doesn't support remote calculation of hashes
 func (o *Object) Hash(r fs.HashType) (string, error) {
-	return "", fs.ErrHashUnsupported
+	session, err := o.fs.sshClient.NewSession()
```

Comment:
However... The cache would need to be invalidated after each operation on the object, which could also happen remotely without warning.